test/integration/targets/aws_ses_identity_dkim/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+---
+# tasks file for aws_ses_identity_dkim test
+
+- name: set up aws connection info
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: yes
+
+- block:
+
+    ## prepare SES idenitiy to run tests on
+
+    - name: Create SES identity for running DKIM module tests on (domain identity)
+      aws_ses_identity:
+        identity: "{{ dkim_identity_domain }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_domain_identity
+
+    - name: Create SES identity for running DKIM module tests on (email identity)
+      aws_ses_identity:
+        identity: "{{ dkim_identity_email }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_email_identity
+
+    ## DKIM enable test
+
+    - name: Enable DKIM verification on domain identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert successful result
+      assert:
+        that:
+          - domain_result.changed == True
+          - domain_result.dkim_enabled == True
+          - domain_result.dkim_tokens|length > 0
+
+    - name: Enable DKIM verification on email identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert successful result
+      assert:
+        that:
+          - email_result.changed == True
+          - email_result.dkim_enabled == True
+          - email_result.dkim_tokens|length > 0
+
+    ## DKIM idempotence test (rerun)
+
+    - name: Enable DKIM verification on domain identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert no change (itempotence)
+      assert:
+        that:
+          - domain_result.changed == False
+          - domain_result.dkim_enabled == True
+
+    - name: Enable DKIM verification on email identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert no change (idempotence)
+      assert:
+        that:
+          - email_result.changed == False
+          - email_result.dkim_enabled == True
+
+    ## DKIM disable test
+
+    - name: Disable DKIM verification on domain identity
```

Comment:
It would be good to add a re-run version of the 2 diable checks as well to confirm idempotence on disable as well as enable.

Reply:
Done